velox/exec/HashTable.h

```diff
@@ -89,6 +89,10 @@ class BaseHashTable {
   struct RowsIterator {
     int32_t hashTableIndex_{-1};
     RowContainerIterator rowContainerIterator_;
+
+    void reset() {
+      *this = {};
```

Comment:
This is an unusual syntax which may confuse future readers. Any particular reason not to explicitly reassign member variables? Perhaps, add a comment to explain what this does?

Reply:
Makes sense. Thank you for clarifying. I don't think we use this syntax elsewhere in the codebase, hence, I thought it might be surprising to readers. Should we start using this more? For example, in OperatorStats::clear.

Reply:
I think we should use it wherever possible (i.e. struct with default constructor that takes no argument).  For `OperatorStats`, there is no default constructor, so the cleaner solution would be calling `clear()` inside constructor and remove the default value on fields.